File: src/backend/search-server/backend/src/models/album.py
from pydantic import BaseModel
from datetime import datetime

from .artist import Artist

class Album(BaseModel):
    id: str
    title: str
    type: str
    releasedAt: str
    artist: Artist

class AlbumIndexRequest(BaseModel):
    id: str
    title: str
    type: str
    releasedAt: str
    artist: Artist

class AlbumNested(BaseModel):
    id: str
    title: str
    type: str
    releasedAt: str

File: src/backend/domain-module/music/src/main/java/com/lalala/music/domain/Album.java
package com.lalala.music.domain;

import java.time.LocalDateTime;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class Album {
    private final Long id;
    private final AlbumType type;
    private final String title;
    private final String coverUrl;
    private final LocalDateTime releasedAt;

    private final Artist artist;
}
